src/context_free.ml

```diff
@@ -440,6 +492,60 @@ let context_free_attribute_modification ~loc =
         "A context-free rule deleted or added attribues of a str/sig item",
       [] )
 
+let handle_attr_replace_once context attrs item base_ctxt : 'a option t =
+  let result =
+    List.find_map attrs ~f:(fun (Rule.Attr_replace.T a) ->
+        let rec get_attr_payloads : type l.
+            ('a, l) Rule.Attribute_list.t ->
+            (bool * l Rule.Parsed_payload_list.t) t = function
+          | [] -> return (false, Rule.Parsed_payload_list.[])
+          | x :: xs ->
+              (if Attribute.Context.equal context (Attribute.context x) then
```

Comment:
If I'm not mistaken, this shouldn't be necessary. The type of `attrs` must match the type of `context`.